src/Shared/EventSource.Testing/TestCounterListener.cs

```diff
@@ -31,14 +37,31 @@ public IAsyncEnumerable<double> GetCounterValues(string counterName, Cancellatio
 
     protected override void OnEventWritten(EventWrittenEventArgs eventData)
     {
-        if (eventData.EventName == "EventCounters")
+        if (eventData.EventSource.Name == _eventSourceName && eventData.EventName == "EventCounters")
         {
             var payload = (IDictionary<string, object>)eventData.Payload[0];
             var counter = (string)payload["Name"];
-            payload.TryGetValue("Increment", out var increment);
-            payload.TryGetValue("Mean", out var mean);
+            if (payload.TryGetValue("Increment", out var increment))
+            {
+                _logger.LogDebug("Counter {CounterName} on event source {EventSourceName} has increment value {Value}.", counter, eventData.EventSource.Name, increment);
+            }
+            if (payload.TryGetValue("Mean", out var mean))
+            {
+                _logger.LogDebug("Counter {CounterName} on event source {EventSourceName} has mean value {Value}.", counter, eventData.EventSource.Name, mean);
+            }
+
+            var value = (double)(increment ?? mean);
             var writer = _counters[counter].Writer;
-            writer.TryWrite((double)(increment ?? mean));
+            writer.TryWrite(value);
+        }
+    }
+
+    public override void Dispose()
+    {
+        base.Dispose();
```

Comment:
Naively, I would have expected the base call to go at the end, though it doesn't appear to matter in this case.

Reply:
Usually, I put base call at the beginning unless there is a reason not to.

Reply:
While I generally agree, I think Dispose is a mirror of the constructor and, for similar reasons, it makes sense to tear down the derived class before tearing down the base class (which it may depend upon).